base/bucket_n1ql.go

```diff
@@ -393,6 +393,8 @@ func isIndexerError(err error) bool {
 		return true
 	} else if IsIndexerRetryBuildError(err) {
 		return true
+	} else if isSelectBucketError(err) {
```

Comment:
This adds a check for this specific error for all queries.  The ticket describes a scenario where we don't do sufficient retry during index initialization on Sync Gateway startup. I feel like a better approach for the problem described on the ticket would be to support retry for all 5000 errors at that point in startup, as opposed to focusing too much on this particular case.

Reply:
Added changes to perform sufficient retry during index initialization on Sync Gateway startup; it supports retry for all 5000 errors at that point in startup instead of checking a particular index error.